dbusmock/templates/systemd-base.py

```diff
@@ -0,0 +1,91 @@
+'''systemd mock template base
+'''
+
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.  See http://www.gnu.org/copyleft/lgpl.html for the full text
+# of the license.
+
+__author__ = 'Jonas Ådahl'
+__copyright__ = '(c) 2021 Red Hat'
+
+from gi.repository import GLib
+
+BUS_NAME = 'org.freedesktop.systemd1'
+MAIN_OBJ = '/org/freedesktop/systemd1'
+MAIN_IFACE = 'org.freedesktop.systemd1.Manager'
+
+
+def load_base(mock, _parameters):
+    mock.next_job_id = 1
+    mock.units = {}
+
+    mock.AddMethod(MAIN_IFACE,
+                   'StartUnit',
+                   'ss', 'o',
+                   'ret = self.StartUnit(self, args[0], args[1])')
```

Comment:
No need to do this, just use the `@dbus.service.method(MAIN_IFACE, in_signature=ss', out_signature='o')` annotation on actual method, and you don't need this at all (or at least no implementation of it).

Reply:
And on other methods as well...

Reply:
That didn't work when the method was implemented in `systemd-base.py`.

Reply:
Oh, I see (even with no implementation at all, right?)... So, maybe you can still use argument expansion though?

Reply:
Argument expansion?

Reply:
```suggestion
                   'ret = self.StartUnit(self, *args)')
```
So you don't have to do it manually for all the calls.

Reply:
Using `*args` now.

> even with no implementation at all, right?)...

Initially I had the decorated methods working without having the implementation in a separate file, but once I moved it to `systemd-base.py` with `systemd-user.py` being the small layer, nothing worked anymore. Tried to find different ways to "merge" the -base.py and the -user.py some how, but gave up and did it like this.

Reply:
Did that.